src/files-action.js

```diff
@@ -35,9 +35,8 @@ if (nextcloudVersionIsGreaterThanOr28) {
 		id: name,
 		displayName: () => t('contacts', 'Import'),
 		default: DefaultType.DEFAULT,
-		mime,
 		enabled: (nodes) => {
-			return nodes.every((node) => node.mime === mime && (node.permissions & Permission.READ))
+			return nodes.some((node) => node.mime !== mime || (node.permissions & Permission.READ) === 0)
```

Comment:
this can't be right. this evaluates to true if there is at least one node with a mismatching mime or no read permission.

this needs a negation, no?

Reply:
right, negation up front
```suggestion
			return !nodes.some((node) => node.mime !== mime || (node.permissions & Permission.READ) === 0)
```

Reply:
Wait, counter argument, considering you do not support `execBatch`, you don't even need to do that.

```suggestion
		if (nodes.length !== 1) {
			return false
		}

		const node = nodes[0]
		return node.mime === mime && (nodes.permissions & Permission.UPDATE) !== 0
```